File: voice-assistant-android/h37a/src/main/java/com/voyah/ai/device/voyah/h37/dc/utils/ReadingLightImpl.java
package com.voyah.ai.device.voyah.h37.dc.utils;

import com.voice.sdk.device.carservice.dc.ReadingLightInterface;

import java.util.ArrayList;

public class ReadingLightImpl implements ReadingLightInterface {

    @Override
    public ArrayList<Integer> getAllPositions() {
        ArrayList<Integer> positionList = new ArrayList<>();
        positionList.add(0);
        positionList.add(1);
        positionList.add(2);
        positionList.add(3);
        positionList.add(4);
        positionList.add(5);
        return positionList;
    }

    @Override
    public boolean isAllCarTTS(ArrayList<Integer> positions) {
        return positions.size() > 4;
    }

}

File: voice-assistant-android/virtual-car/src/main/java/com/voyah/ai/device/carservice/ReadingLightImpl.kt
package com.voyah.ai.device.carservice

import com.voice.sdk.device.carservice.dc.ReadingLightInterface
import java.util.ArrayList

/**
 * @Date 2025/4/1 15:56
 * @Author 8327821
 * @Email *
 * @Description .
 **/
object ReadingLightImpl: ReadingLightInterface {
    override fun getAllPositions(): ArrayList<Int> {
        return arrayListOf(0,1,2,3,4,5)
    }

    override fun isAllCarTTS(p0: ArrayList<Int>?): Boolean {
        return if (p0 == null) {
            false
        } else {
            p0.size > 4
        }
    }
}